### ISSUE TITLE
IPZN and the future of ZeroNetJS

### ISSUE THREAD

**Issue by mkg20001** [ZeroNetJS/zeronet-js#109]
A few years ago, I started this project as a tiny summer holiday free-time-project, but had to give up mainly due to lack of active contributors and the fact that I still had school at that time.

Recently, @blurHY has contacted me about his ideas of creating [IPZN](https://github.com/ipzn), which seem to align very well with those of this project.

The existing ZeroNetJS code will likely serve as a ground-base for building the InterPlanetary ZeroNet client, which will utilize both the existing ZeroNet network, as well as the in-development IPLD integration, which will allow us to do things way beyond what ZeroNet can do, possibly leading to a completely new ZeroNet-Network.

Among others, the IPLD/IPFS network could add capapbilities such as immutability (permanent, verifiable archival), interoperability (imagine github on ZeroNet, but with push access - this is partly already possible with IPLD-git), and security (by attaching block-hashes from well-known cryptocurrencies to posts to improve timestamp reliability, to allow for more complex permission management)

The future plans are being discussed, but as far as I can tell, this definitivly has the potential to be something big

// cc @kustomzone  @DaniellMesquita @blurHY @MuxZeroNet @ZeroNetJS/devs

**Comment by blurHY** [comment]
Indeed, zeronet, ipfs and blockchain are potential. Ethereum is an example, but it's not focused on anti-censorship. 

I'm a student too, interestingly. 

Plan/Wiki/How it works: https://gitlab.com/ipzn/ipzn/wikis/home

PS: I've configured the auto-mirror from gitlab to github

**Comment by blurHY** [comment]
Do you think we should invite more people here ? 

@d14na They seems interested in this project.
@HelloZeroNet

**Comment by mkg20001** [comment]
> I'm a student too, interestingly.

I used to be. Until very recently. Now I'm a freelancer :P

**Comment by d14na** [comment]
i am very interested. my current commitments should be wrapping up in the next few weeks, and i'm planning to get back to 0net development full-time thru the end of the year.

**Comment by d14na** [comment]
Just ran into this page again. How you guys doing? I see a lot of discussion and planning, but Is anyone working on code? I'm free this weekend and mon is a holiday, so I'm looking to get into some code, lol.
